Lib/test/test_ast.py

```diff
@@ -2347,6 +2348,130 @@ def visit_Ellipsis(self, node):
         ])
 
 
+class NodeVisitorTests(BaseNodeVisitorCases, unittest.TestCase):
+    visitor_class = ast.NodeVisitor
+
+
+class NodeTransformerTests(BaseNodeVisitorCases, unittest.TestCase):
+    visitor_class = ast.NodeTransformer
+    maxDiff = None  # `ast.dump` might generate a large string
+
+    def assertASTTransformation(self, tranformer_class,
+                                initial_code, expected_code):
+        initial_ast = ast.parse(dedent(initial_code))
+        expected_ast = ast.parse(dedent(expected_code))
+
+        tranformer = tranformer_class()
+        result_ast = ast.fix_missing_locations(tranformer.visit(initial_ast))
+
+        self.assertEqual(ast.dump(result_ast),
```

Comment:
IIRC we have a more performant way to compare AST trees. Check out  `traverse_compare` in `test_unparse`

Reply:
Done! 👍